File: Capitulo-08/Algoritimos_Gulosos/Exercicios/Python/main.py
def greedy_coin_change(coins, amount):
    coins.sort(reverse=True)

    result = []

    for coin in coins:
        while amount >= coin:
            amount -= coin
            result.append(coin)

    return result


coins = [25, 10, 5, 1]
amount = 63

print(greedy_coin_change(coins, amount))

File: Capitulo-08/Algoritimos_Gulosos/Exercicios/Ruby/main.rb
def greedy_coin_change(coins, amount)
  coins.sort!.reverse!
  result = []

  coins.each do |coin|
    while amount >= coin
      amount -= coin
      result << coin
    end
  end

  result
end

coins = [25, 10, 5, 1]
amount = 63
puts greedy_coin_change(coins, amount).inspect